src/gc-stock.h

```diff
@@ -319,12 +373,47 @@ STATIC_INLINE void gc_alloc_map_maybe_create(char *_data) JL_NOTSAFEPOINT
     }
 }
 
-// Page layout:
-//  Metadata pointer: sizeof(jl_gc_pagemeta_t*)
-//  Padding: GC_PAGE_OFFSET - sizeof(jl_gc_pagemeta_t*)
-//  Blocks: osize * n
-//    Tag: sizeof(jl_taggedvalue_t)
-//    Data: <= osize - sizeof(jl_taggedvalue_t)
+/*
+ * Page Layout
+ *
+ * Each pool-allocated page is divided into three main sections:
+ *
+ * - Metadata Pointer
+ *   - Size: sizeof(jl_gc_pagemeta_t*)
+ *   - Points to the page metadata structure.
+ *
+ * - Padding
+ *   - Size: GC_PAGE_OFFSET - sizeof(jl_gc_pagemeta_t*)
+ *   - Ensures proper alignment of the blocks.
+ *
+ * - Blocks
+ *   - Size per block: osize
+ *   - Each block consists of:
+ *     - Tag: sizeof(jl_taggedvalue_t)
+ *     - Data: up to (osize - sizeof(jl_taggedvalue_t))
+ *
+ * Example layout:
+ *
+ *   +----------------------+ <- page start
+ *   | Metadata Pointer     |  sizeof(jl_gc_pagemeta_t*)
+ *   +----------------------+
+ *   | Padding              |  GC_PAGE_OFFSET - sizeof(jl_gc_pagemeta_t*)
+ *   +----------------------+ <- GC_PAGE_OFFSET
+ *   | Block 0              |  osize
+ *   |   +----------------+|
+ *   |   | Tag            || sizeof(jl_taggedvalue_t)
+ *   |   +----------------+|
+ *   |   | Data           || <= osize - sizeof(jl_taggedvalue_t)
+ *   |   +----------------+|
+ *   | Block 1              |  osize
+ *   |   +----------------+|
+ *   |   | Tag            || sizeof(jl_taggedvalue_t)
+ *   |   +----------------+|
+ *   |   | Data           || <= osize - sizeof(jl_taggedvalue_t)
+ *   |   +----------------+|
+ *   | ...                  |
+ *   +----------------------+ <- page end
```

Comment:
Alignment is off, or is that meant to be so?

Reply:
I'm talking about the alignment of the vertical lines on the right of the diagram:

<img width="42" height="418" alt="image" src="https://github.com/user-attachments/assets/a2951fc7-8fdf-4688-b6a1-d75c7bb1a5ee" />

I see the same also in my local editor, so it isn't a problem of GitHub rendering.